middleman_protocol/middleman_protocol/helpers.py

```diff
@@ -0,0 +1,18 @@
+from asyncio import StreamReader
+
+from middleman_protocol.constants import ErrorCode
+from middleman_protocol.constants import FRAME_SEPARATOR
+from middleman_protocol.constants import MIDDLEMAN_EXCEPTION_TO_ERROR_CODE_MAP
+from middleman_protocol.exceptions import MiddlemanProtocolError
+from middleman_protocol.stream import escape_decode_raw_message
+
+
+async def handle_frame_receive_async(reader: StreamReader) -> bytes:
+    raw_data = await reader.readuntil(FRAME_SEPARATOR)
+    index = raw_data.index(FRAME_SEPARATOR)
+    raw_data_without_separator = raw_data[:index]
+    return escape_decode_raw_message(raw_data_without_separator)
+
+
```

Comment:
There should be a function which does opposite of `send_over_stream`.

Reply:
there is `handle_frame_receive_async` - it does the opposite: receives frame asynchronously

Reply:
Oh sorry, by opposite I have meant `async` version, but I see there is already something like `send_over_stream_async`.